### PR TITLE
Feat/s4 admin get post ou 81 phat

### PR DESCRIPTION
**Pull request by tranducphat204**

### CODE REVIEW

File: src/post/schemas/post.schema.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export class Post {
 
     @Prop()
     likes: [];
+  id: any;
```
Comment (NhatNguyen1502) [inline src/post/schemas/post.schema.ts]: delete this Prop, we dont need declare this prop. read this (https://stackoverflow.com/questions/71190267/nest-js-how-to-correctly-set-id-property-in-your-schema) to know more

Comment (tranducphat204) [reply src/post/schemas/post.schema.ts]: I have edited according to your feedback

File: src/post/post.controller.ts [hunk 2]
```diff
@@ -36,25 +36,44 @@ export class PostController {
...
+        message: 'Error: Unable to fetch posts',
+        error,
+      });
+    }
   }
 
   @Get(':id')
-  findOne(@Param('id') id: string) {
-    return this.postService.findOne(+id);
+  @ApiOperation({ summary: 'Get post by ID' })
```
Comment (NhatNguyen1502) [inline src/post/post.controller.ts:55]: add api response decorator

File: src/post/post.controller.ts [hunk 3]
```diff
@@ -52,16 +53,18 @@ export class PostController {
 
   @Get(':id')
   @ApiOperation({ summary: 'Get post by ID' })
-  async findOne(@Param('id') id: string, @Res() res) {
+  @ApiOkResponse({ description: 'Get post by ID successfully!!!' })
+  @ApiBadRequestResponse({ description: 'Bad request. Check your input data!' })
```
Comment (NhatNguyen1502) [inline src/post/post.controller.ts]: not good description

**Review by NhatNguyen1502 — APPROVED** [review]

**Pull request merged**